### PR TITLE
feat: search by author

### PR DESCRIPTION
**Pull request by iliapolo**
In the result cards, clicking on the author will now result in a search query for that author, instead of going to the author UR, which will now only be available from the operator area of the package page. 

Closes https://github.com/cdklabs/construct-hub-webapp/issues/271

### CODE REVIEW

**Review by MrArnoldPalmer — APPROVED** [review]

**Review by gabewomble — CHANGES_REQUESTED** [review]
Wait

File: src/components/CatalogCard/CatalogCard.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import {
   Divider,
   Flex,
   LinkBox,
+  Link as UILnk,
```
Comment (gabewomble) [inline src/components/CatalogCard/CatalogCard.tsx:6]: typo

File: src/components/CatalogCard/CatalogCard.tsx [hunk 2]
```diff
@@ -140,24 +142,14 @@ export const CatalogCard: FunctionComponent<CatalogCardProps> = ({
...
-            <Text data-testid={testIds.author} fontSize="sm" isTruncated>
-              {typeof author === "string" ? author : author.name}
-            </Text>
-          )}
+          <UILnk
+            color="blue.500"
+            fontSize="sm"
+            href={`./search?q=${authorName}`}
+            onClick={(e) => e.stopPropagation()}
+          >
```
Comment (gabewomble) [inline src/components/CatalogCard/CatalogCard.tsx:150]:
```suggestion
          <UILink
            as={Link}
            color="blue.500"
            fontSize="sm"
            to={getSearchPath({ query: authorName })}
          >
```

Comment (gabewomble) [reply src/components/CatalogCard/CatalogCard.tsx:150]: getSearchPath will need to be imported from the same file as `getPackagePath` (which is already imported)

Comment (iliapolo) [reply src/components/CatalogCard/CatalogCard.tsx]: Done

File: src/components/CatalogCard/CatalogCard.tsx (hunk 1 above)
Comment (iliapolo) [reply src/components/CatalogCard/CatalogCard.tsx]: Danke

**Review by gabewomble — APPROVED** [review]

**Pull request merged**